hi-ml-multimodal/src/health_multimodal/image/utils.py

```diff
@@ -0,0 +1,53 @@
+#  -------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  -------------------------------------------------------------------------------------------
+
+import tempfile
+from pathlib import Path
+
+from torchvision.datasets.utils import download_url
+
+from .. import REPO_URL
+from .model import ImageModel
+from .inference_engine import ImageInferenceEngine
+from .data.transforms import create_chest_xray_transform_for_inference
+
+IMAGE_WEIGHTS_NAME = ""  # TODO
+IMAGE_WEIGHTS_URL = f"{REPO_URL}/raw/main/{IMAGE_WEIGHTS_NAME}"
+IMAGE_WEIGHTS_MD5 = ""  # TODO
+
+
+def _download_image_model_weights() -> Path:
+    """Download image model weights from Hugging Face."""
+    root_dir = tempfile.gettempdir()
+    download_url(
+        IMAGE_WEIGHTS_URL,
+        root=tempfile.gettempdir(),
+        filename=IMAGE_WEIGHTS_NAME,
+        md5=IMAGE_WEIGHTS_MD5,
+    )
+    return Path(root_dir, IMAGE_WEIGHTS_NAME)
+
+
+def get_cxr_resnet(model_type: str = "resnet50", joint_feature_size: int = 128) -> ImageModel:
+    """Download weights from Hugging Face and instantiate the image model."""
+    resnet_checkpoint_path = _download_image_model_weights()
+    image_model = ImageModel(
+        img_model_type=model_type,
```

Comment:
The checkpoint is only for `resnet50` so perhaps there is not much point in parameterising it here because the feature size and architecture type are fixed. If any of them are changed, there will be a run-time error.